pkg/minikube/cluster/status.go

```diff
@@ -0,0 +1,510 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cluster
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+	"github.com/docker/machine/libmachine"
+	"github.com/docker/machine/libmachine/state"
+	"github.com/pkg/errors"
+	"k8s.io/klog/v2"
+	"k8s.io/minikube/pkg/minikube/bootstrapper/bsutil/kverify"
+	"k8s.io/minikube/pkg/minikube/config"
+	"k8s.io/minikube/pkg/minikube/constants"
+	"k8s.io/minikube/pkg/minikube/driver"
+	"k8s.io/minikube/pkg/minikube/kubeconfig"
+	"k8s.io/minikube/pkg/minikube/localpath"
+	"k8s.io/minikube/pkg/minikube/machine"
+	"k8s.io/minikube/pkg/minikube/out/register"
+	"k8s.io/minikube/pkg/version"
+)
+
+// Additional legacy states
+const (
+	// Configured means configured
+	Configured = "Configured" // ~state.Saved
+	// Misconfigured means misconfigured
+	Misconfigured = "Misconfigured" // ~state.Error
+	// Nonexistent means the resource does not exist
+	Nonexistent = "Nonexistent" // ~state.None
+	// Irrelevant is used for statuses that aren't meaningful for worker nodes
+	Irrelevant = "Irrelevant"
+)
+
+// New status modes, based roughly on HTTP/SMTP standards
+const (
+
+	// 1xx signifies a transitional state. If retried, it will soon return a 2xx, 4xx, or 5xx
+
+	Starting  = 100
+	Pausing   = 101
+	Unpausing = 102
+	Stopping  = 110
+	Deleting  = 120
+
+	// 2xx signifies that the API Server is able to service requests
+
+	OK       = 200
+	OKHAppy  = 201
+	Warning  = 203
+	Degraded = 204
+
+	// 4xx signifies an error that requires help from the client to resolve
+
+	NotFound = 404
+	Stopped  = 405
+	Paused   = 418 // I'm a teapot!
+
+	// 5xx signifies a server-side error (that may be retryable)
+
+	Error               = 500
+	InsufficientStorage = 507
+	Unknown             = 520
+)
+
+var (
+	exitCodeToHTTPCode = map[int]int{
+		// exit code 26 corresponds to insufficient storage
+		26: 507,
+	}
+
+	codeNames = map[int]string{
+		100: "Starting",
+		101: "Pausing",
+		102: "Unpausing",
+		110: "Stopping",
+		103: "Deleting",
+
+		200: "OK",
+		201: "OKHAppy",
+		203: "Warning",
+		204: "Degraded",
+
+		404: "NotFound",
+		405: "Stopped",
+		418: "Paused",
+
+		500: "Error",
+		507: "InsufficientStorage",
+		520: "Unknown",
+	}
+
+	codeDetails = map[int]string{
+		507: "/var is almost out of disk space",
+	}
+)
+
+// Status holds string representations of component states
+type Status struct {
+	Name       string
+	Host       string
+	Kubelet    string
+	APIServer  string
+	Kubeconfig string
+	Worker     bool
+	TimeToStop string `json:",omitempty"`
+	DockerEnv  string `json:",omitempty"`
+	PodManEnv  string `json:",omitempty"`
+}
+
+// ClusterState holds a cluster state representation
+//
+//nolint:revive
+type ClusterState struct {
```

Comment:
```suggestion
// State holds a cluster state representation
//
//nolint:revive
type State struct {
```

Since this is already in the `cluster` package, `cluster` prefixes should be removed to prevent stuttering.

ie. `cluster.ClusterState` instead `cluster.State`